test/kw/kw_renew_token.h

```diff
@@ -0,0 +1,39 @@
+/**
+ * \file kw_renew_token.h
+ *
+ * \copyright Copyright (C) 2019 Kano Computing Ltd.
+ *            License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+ *
+ * \brief Tests for the Mercury::KW - Kano World - functions
+ *
+ */
+
+#ifndef TEST_KW
+#define TEST_KW
+
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <mercury/kw/kw.h>
+
+using testing::Eq;
+
+TEST(kw, renew_token)
+{
+    KanoWorld kw;
```

Comment:
An idea to allow you to do dependency injection here might be that you create a `HTTPClient` interface with `get()`, `post()`, etc methods which you can create an implementation of (something like `LibCurlClient`) which simply translates into `libCurl` language. This can then be passed in when you instantiate a `KanoWorld` object and then you can create a mock of the interface in the tests which replies with the test data when you call `get()` with the right arguments.

The pattern should look something like this (very broadly):

```c++
class HTTPClient {
public:
    Response get(...) = 0;
    void post(...) = 0;
}

class LibCurlClient : public HTTPClient {
public:
    Response get(...) {
        // GET with libCurl
    }
    void post(...) {
        // POST with libCurl
    }
}

class HTTPMock: public HTTPClient {
  MOCK_METHOD1(get, Response(...));
};

TEST(kw, renew_token)
{
    HTTPMock mockClient;
    EXPECT_CALL(mockClient, Response(/* args matcher */))
        .WillRepeatedly(Return(/* test data */));

    KanoWorld kw(mockClient);
    EXPECT_EQ(kw.renew_token("1234567890"), true);
}
```